docs/source/spawners.md

```diff
@@ -162,7 +162,32 @@ When `Spawner.start` is called, this dictionary is accessible as `self.user_opti
 
 [Spawner]: https://github.com/jupyter/jupyterhub/blob/master/jupyterhub/spawner.py
 
-
 ## Writing a custom spawner
 
 If you are interested in building a custom spawner, you can read [this tutorial](http://jupyterhub-tutorial.readthedocs.io/en/latest/spawners.html).
+
+## Exposing resource limits & guarantees
+
+Some spawners allow setting CPU / Memory limits and / or guarantees on the spawned single-user servers. Such spawners should expose the limits/guarantees to the single-user servers via environment variables.
+
+### Limits
+
+If limits are being enforced, the following environment variables should be set in the single-user notebooks' environment:
+
+* `LIMIT_MEM`: Maximum amount of memory the single user notebook is allowed to use, specified in integer bytes.
+* `LIMIT_CPU`: Maximum number of 'cpu cores' a single user notebook is allowed to use, specified as a float. For example, `1.0` allows you to use one CPU core, `0.5` half a CPU core, `4.0` four CPU cores, etc.
+
+If no resource limits are being enforced, these env variables should just not be set. Limits are also not guarantees - there are no guarantees that a single user notebook can use all the resources up to its limit.
```

Comment:
It's important to note that the a limit is not a guarantee. If a limit is set, for example, `LIMIT_CPU` is `2.0` CPU cores, there is no guarantee that a single-user notebook server will be provided the maximum of `2.0` CPU cores.  Instead, the single-user notebook server can be given up to `2.0` CPU cores, the CPU maximum limit.